db.go

```diff
@@ -434,21 +435,57 @@ func (db *DB) Close() error {
 
 	db.closed = true
 
+	GCEnable := db.opt.GCWhenClose
+
+	err := db.release()
+	if err != nil {
+		return err
+	}
+
+	if GCEnable {
+		runtime.GC()
+	}
+
+	return nil
+}
+
+// release set all obj in the db instance to nil
+func (db *DB) release() error {
 	err := db.ActiveFile.rwManager.Release()
 	if err != nil {
 		return err
 	}
 
+	db.BPTreeIdx = nil
+
+	db.BPTreeKeyEntryPosMap = nil
+
+	db.bucketMetas = nil
+
+	db.SetIdx = nil
+
+	db.SortedSetIdx = nil
+
+	db.Index = nil
+
 	db.ActiveFile = nil
 
-	db.BPTreeIdx = nil
+	db.ActiveBPTreeIdx = nil
+
+	db.ActiveCommittedTxIdsIdx = nil
+
+	db.committedTxIds = nil
 
 	err = db.fm.close()
 
 	if err != nil {
 		return err
 	}
 
+	db.fm = nil
+
+	db = nil
+
```

Comment:
I believe you should check if the value of GCWhenClose is true here, then if so, you should call runtime.GC() method here.